src/scheduler/server_info.cpp

```diff
@@ -776,18 +776,21 @@ query_server_dyn_res(server_info *sinfo)
 					log_eventf(PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
 					"Program %s timed out", dr.command_line.c_str());
 				}
-				if (pid > 0 && ret > 0) {
-					/* Parent; only open if child created and select showed sth to read,
-					 * but assume fdopen can't fail
-					 */
-					fp = fdopen(pdes[0], "r");
+				if (pid > 0) {
 					close(pdes[1]);
-					if (fgets(buf, sizeof(buf), fp) == NULL) {
-						pipe_err = errno;
+					if (ret > 0) {
+						/* Parent; only open if child created and select showed sth to read,
+						* but assume fdopen can't fail
+						*/
+						fp = fdopen(pdes[0], "r");
+						if (fgets(buf, sizeof(buf), fp) == NULL) {
+							pipe_err = errno;
+						} else
+							k = strlen(buf);
+						if (fp != NULL)
+							fclose(fp);
 					} else
-						k = strlen(buf);
-					if (fp != NULL)
-						fclose(fp);
+						close(pdes[0]);
```

Comment:
should pdes[0] be closed all the time? not just when ret is <= 0?

Reply:
pdes[0] is fp, so when you fclose(fp), it should close pdes[0].

Reply:
Duh, of course.